serializers.ts

```diff
@@ -79,12 +79,23 @@ function parseMessage(
   // Read the metadata
   const metadataUint8Array = new Uint8Array(buffer, 4, metadataLength);
   const metadataString = new TextDecoder().decode(metadataUint8Array);
-  const metadata = JSON.parse(metadataString);
 
   // Read the binary data
   const binaryData = new Uint8Array(buffer, 4 + metadataLength);
+  try {
+    const metadata = JSON.parse(metadataString);
 
-  return { metadata, binaryData };
+    return { metadata, binaryData };
+  } catch (error) {
+    console.log({
+      error,
+      buffer,
+      binaryData,
+      metadataString,
+      metadataUint8Array,
+    });
+    return { metadata: "", binaryData };
```

Comment:
I'd say metadata can't be a string but rather an object, however there's no way to parse data without metadata information, so my preference is to rethrow the error instead of creating inconsistent behaviours returning empty strings and arbitrary data.

Reply:
```suggestion
   throw error;
```